modules/ensemble/lib/framework/apiproviders/firebase_functions/firebase_functions_api_provider.dart

```
@@ -0,0 +1,485 @@
+  }
+
+  /// Get Firebase options from config map (similar to Firestore approach)
+  static Future<FirebaseOptions?> _getFirebaseOptions(Map<String, dynamic>? config) async {
+    try {
+      if (config == null) {
+        log('No config provided, using default Firebase initialization');
+        return null;
+      }
+
+      // Create a FirebaseConfig-like structure
+      FirebaseOptions? platformOptions;
+      
+      if (defaultTargetPlatform == TargetPlatform.iOS) {
+        // Get iOS configuration from config
+        var iosConfig = config['ios'];
+        if (iosConfig != null) {
+          platformOptions = FirebaseOptions(
+            apiKey: iosConfig['apiKey'],
+            appId: iosConfig['appId'],
+            messagingSenderId: iosConfig['messagingSenderId'],
+            projectId: iosConfig['projectId'],
+          );
+        }
+      } else if (defaultTargetPlatform == TargetPlatform.android) {
+        // Get Android configuration from config
+        var androidConfig = config['android'];
+        if (androidConfig != null) {
+          platformOptions = FirebaseOptions(
+            apiKey: androidConfig['apiKey'],
+            appId: androidConfig['appId'],
+            messagingSenderId: androidConfig['messagingSenderId'],
+            projectId: androidConfig['projectId'],
+          );
+        }
+      } else if (kIsWeb) {
+        // Get Web configuration from config
+        var webConfig = config['web'];
+        if (webConfig != null) {
+          platformOptions = FirebaseOptions(
+            apiKey: webConfig['apiKey'],
+            appId: webConfig['appId'],
+            messagingSenderId: webConfig['messagingSenderId'],
+            projectId: webConfig['projectId'],
+            authDomain: webConfig['authDomain'],
+            storageBucket: webConfig['storageBucket'],
+            measurementId: webConfig['measurementId'],
+          );
+        }
+      }
+
+      // Fallback to hardcoded values if no config found
```

But why?? Shouldn't it throw error instead of adding hardcoded values of a random Firebase account?